### PR TITLE
refactor(datasets): rename legacy datasets to inferences

### LINKED ISSUE #3171: [gql] migrate dataset to inferences

### PR DESCRIPTION
**Pull request by mikeldking**
resolves #3171

### CODE REVIEW

File: app/src/store/pointCloudStore.ts [hunk 1]
```diff
@@ -320,10 +320,10 @@ export interface PointCloudProps {
    */
   coloringStrategy: ColoringStrategy;
   /**
-   * The visibility of the two datasets in the point cloud.
+   * The visibility of the two inferencess in the point cloud.
    * @default { primary: true, reference: true }
    */
-  datasetVisibility: DatasetVisibility;
+  inferencesVisibility: DatasetVisibility;
```
Comment (axiomofjoy) [inline app/src/store/pointCloudStore.ts:326]: Missed `DatasetVisibility`.

File: app/src/store/pointCloudStore.ts [hunk 2]
```diff
@@ -1249,11 +1249,11 @@ async function fetchPointEvents(eventIds: string[]): Promise<PointDataMap> {
                 prompt
                 response
               }
-              # TODO: delineate between corpus events and dataset events
+              # TODO: delineate between corpus events and inferences events
```
Comment (axiomofjoy) [inline app/src/store/pointCloudStore.ts:1252]: Can we nix this todo?

File: tests/server/api/types/test_inference.py [hunk 3]
```diff
@@ -12,7 +12,7 @@
 
 
 @pytest.fixture
-def input_dataset() -> InternalInference:
+def input_inferences() -> InternalInference:
```
Comment (axiomofjoy) [inline tests/server/api/types/test_inference.py:15]:
```suggestion
def input_inferences() -> InternalInferences:
```

File: tests/server/api/types/test_inference.py [hunk 4]
```diff
@@ -34,21 +34,21 @@ def input_dataset() -> InternalInference:
 class TestDatasetEvents:
```
Comment (axiomofjoy) [inline tests/server/api/types/test_inference.py:34]:
```suggestion
class TestInferenceEvents:
```

File: src/phoenix/trace/trace_dataset.py
Comment (axiomofjoy) [inline src/phoenix/trace/trace_dataset.py:1]: Do we need to change the name of this file?

File: src/phoenix/trace/trace_dataset.py [hunk 5]
```diff
@@ -280,7 +280,9 @@ def load(
         """
         if not isinstance(id, UUID):
             id = UUID(id)
-        path = Path(directory or TRACE_DATASET_DIR) / TRACE_DATASET_PARQUET_FILE_NAME.format(id=id)
+        path = Path(directory or TRACE_INFERENCES_DIR) / TRACE_DATASET_PARQUET_FILE_NAME.format(
+            id=id
```
Comment (axiomofjoy) [inline src/phoenix/trace/trace_dataset.py:284]: I'm not clear on why we're keeping one name but not the other.

File: src/phoenix/core/model_schema.py [hunk 6]
```diff
@@ -746,14 +746,14 @@ class Model:
     a column of NaNs.
     """
 
-    _datasets: Dict[DatasetRole, Dataset]
+    _datasets: Dict[InferencesRole, Inferences]
```
Comment (axiomofjoy) [inline src/phoenix/core/model_schema.py:749]: Missed `_datasets`

File: src/phoenix/core/model_schema.py [hunk 7]
```diff
@@ -786,8 +786,8 @@ def __init__(
         padded_dfs = _add_padding(str_col_dfs, pd.DataFrame)
         padded_df_names = _add_padding(df_names, _rand_str)
         datasets = starmap(
```
Comment (axiomofjoy) [inline src/phoenix/core/model_schema.py:788]: Missed `datasets`.

File: src/phoenix/core/model_schema.py [hunk 8]
```diff
@@ -786,8 +786,8 @@ def __init__(
...
-            self._new_dataset,
-            zip(padded_dfs, padded_df_names, DatasetRole),
+            self._new_inferences,
+            zip(padded_dfs, padded_df_names, InferencesRole),
         )
         # Store datasets by role.
         object.__setattr__(
```
Comment (axiomofjoy) [inline src/phoenix/core/model_schema.py:793]: Missed here and just below.

File: src/phoenix/core/model_schema.py [hunk 9]
```diff
@@ -921,29 +921,29 @@ def export_rows_as_parquet_file(
...
         Parameters
         ----------
-        row_numbers: Mapping[DatasetRole, Iterable[int]]
+        row_numbers: Mapping[InferencesRole, Iterable[int]]
             mapping of dataset role to list of row numbers
         parquet_file: file handle
             output parquet file handle
-        cluster_ids: Optional[Mapping[DatasetRole, Mapping[int, str]]]
+        cluster_ids: Optional[Mapping[InferencesRole, Mapping[int, str]]]
             mapping of dataset role to mapping of row number to cluster id.
```
Comment (axiomofjoy) [inline src/phoenix/core/model_schema.py:929]:
```suggestion
            mapping of inferences role to mapping of row number to cluster id.
```

File: src/phoenix/core/model_schema.py [hunk 10]
```diff
@@ -921,29 +921,29 @@ def export_rows_as_parquet_file(
 
         Parameters
         ----------
-        row_numbers: Mapping[DatasetRole, Iterable[int]]
+        row_numbers: Mapping[InferencesRole, Iterable[int]]
             mapping of dataset role to list of row numbers
```
Comment (axiomofjoy) [inline src/phoenix/core/model_schema.py:925]:
```suggestion
            mapping of inferences role to list of row numbers
```

File: src/phoenix/core/model_schema.py [hunk 11]
```diff
@@ -921,29 +921,29 @@ def export_rows_as_parquet_file(
...
             output parquet file handle
-        cluster_ids: Optional[Mapping[DatasetRole, Mapping[int, str]]]
+        cluster_ids: Optional[Mapping[InferencesRole, Mapping[int, str]]]
             mapping of dataset role to mapping of row number to cluster id.
             If cluster_ids is non-empty, a new column is inserted to the
             dataframe containing the cluster IDs of each row in the exported
             data. The name of the added column name is `__phoenix_cluster_id__`.
         """
         export_dataframes = [pd.DataFrame()]
         model_has_multiple_datasets = sum(not df.empty for df in self._datasets.values()) > 1
```
Comment (axiomofjoy) [inline src/phoenix/core/model_schema.py:935]:
```suggestion
        model_has_multiple_inferences = sum(not df.empty for df in self._datasets.values()) > 1
```

File: src/phoenix/core/model_schema_adapter.py
Comment (axiomofjoy) [inline src/phoenix/core/model_schema_adapter.py:1]: Missing `DatasetName`

File: src/phoenix/core/model_schema_adapter.py [hunk 12]
```diff
@@ -17,7 +17,7 @@
 DisplayName: TypeAlias = str
 
 
-def create_model_from_datasets(*datasets: Optional[Inferences]) -> Model:
+def create_model_from_inferences(*datasets: Optional[Inferences]) -> Model:
```
Comment (axiomofjoy) [inline src/phoenix/core/model_schema_adapter.py:20]: Missing argument

File: app/src/components/pointcloud/EventItem.tsx [hunk 13]
```diff
@@ -517,8 +524,8 @@ function EventItemFooter({
   color,
   group,
   showDataset,
-  datasetName,
-}: Pick<EventItemProps, "group" | "color" | "datasetName"> & {
+  inferencesName,
+}: Pick<EventItemProps, "group" | "color" | "inferencesName"> & {
```
Comment (axiomofjoy) [inline app/src/components/pointcloud/EventItem.tsx:528]: Missed `showDataset`

File: app/src/store/pointCloudStore.ts (hunk 2 above)
Comment (mikeldking) [reply app/src/store/pointCloudStore.ts:1252]: I think I need to keep this

File: app/src/store/pointCloudStore.ts (hunk 1 above)
Comment (mikeldking) [reply app/src/store/pointCloudStore.ts]: Ooof, thanks

File: src/phoenix/server/api/types/Model.py [hunk 14]
```diff
@@ -194,17 +194,17 @@ def performance_time_series(
...
-        ] = DatasetRole.primary,
+        ] = InferencesRole.primary,
     ) -> PerformanceTimeSeries:
-        if not isinstance(dataset_role, DatasetRole):
-            dataset_role = DatasetRole.primary
+        if not isinstance(inferences_role, InferencesRole):
+            inferences_role = InferencesRole.primary
         model = info.context.model
-        dataset = model[dataset_role.value]
+        dataset = model[inferences_role.value]
```
[the remark this answers is not in the document]
Comment (mikeldking) [reply src/phoenix/server/api/types/Model.py:207]: Done

File: app/src/components/pointcloud/DatasetVisibilitySettings.tsx
[the remark this answers is not in the document]
Comment (mikeldking) [reply app/src/components/pointcloud/DatasetVisibilitySettings.tsx:1]: Fixed

File: src/phoenix/core/model_schema_adapter.py (hunk 12 above)
Comment (mikeldking) [reply src/phoenix/core/model_schema_adapter.py]: made it inference_sets

File: app/src/@types/inferences.d.ts [hunk 15]
```diff
@@ -0,0 +1,5 @@
+/**
+ * The role of inferences. E.g. primary is the inferences under test, reference is the
+ * inferences used for comparison.
+ */
```
Comment (axiomofjoy) [inline app/src/@types/inferences.d.ts:4]:
```suggestion
/**
 * The role of inferences. E.g., primary refers to the inferences under test, reference refers to the
 * inferences used for comparison.
 */
```

File: app/src/contexts/InferencesContext.tsx [hunk 16]
```diff
@@ -0,0 +1,62 @@
...
+  referenceInferences: InferencesDef | null;
+  corpusInferences: InferencesDef | null;
+  getDatasetNameByRole: (role: InferencesRole) => string;
+};
+
+export const InferencesContext = createContext<InferencesContextType | null>(
+  null
+);
+
+export function useDatasets() {
```
Comment (axiomofjoy) [inline app/src/contexts/InferencesContext.tsx:23]: Missed here.

File: app/src/contexts/InferencesContext.tsx [hunk 17]
```diff
@@ -0,0 +1,62 @@
...
+};
+
+export function InferencesProvider(props: InferencesProviderProps) {
+  return (
+    <InferencesContext.Provider
+      value={{
+        primaryInferences: props.primaryInferences,
+        referenceInferences: props.referenceInferences,
+        corpusInferences: props.corpusInferences,
+        getDatasetNameByRole: (inferencesRole: InferencesRole) => {
```
Comment (axiomofjoy) [inline app/src/contexts/InferencesContext.tsx:45]: Missed here

File: app/src/constants/pointCloudConstants.tsx
Comment (axiomofjoy) [inline app/src/constants/pointCloudConstants.tsx:1]: need to rename `DatasetGroup` in this file,

File: app/src/pages/embedding/PointSelectionGrid.tsx [hunk 18]
```diff
@@ -72,8 +72,8 @@ export function PointSelectionGrid(props: PointSelectionGridProps) {
             data?.embeddingMetadata ?? {};
           const { predictionLabel = null, actualLabel = null } =
             data?.eventMetadata ?? {};
-          const datasetRole = getDatasetRoleFromEventId(event.id);
-          const datasetName = getDatasetNameByRole(datasetRole);
+          const inferencesRole = getInferencesRoleFromEventId(event.id);
+          const inferencesName = getDatasetNameByRole(inferencesRole);
```
Comment (axiomofjoy) [inline app/src/pages/embedding/PointSelectionGrid.tsx:76]: Missed `getDatasetNameByRole`

File: app/src/pages/embedding/PointSelectionTable.tsx [hunk 19]
```diff
@@ -32,7 +32,7 @@ export function PointSelectionTable({
   data: ModelEvent[];
   onPointSelected: (pointId: string) => void;
 }) {
-  const { primaryDataset, referenceDataset } = useDatasets();
+  const { primaryInferences, referenceInferences } = useDatasets();
```
Comment (axiomofjoy) [inline app/src/pages/embedding/PointSelectionTable.tsx:35]: Missed `useDatasets`

File: app/src/components/pointcloud/PointCloudPointTooltip.tsx [hunk 20]
```diff
@@ -42,8 +42,8 @@ export const PointCloudPointTooltip = () => {
   if (baseEvent == null || eventDetails == null) return null;
   const group = eventIdToGroup[hoveredEventId];
   const color = pointGroupColors[eventIdToGroup[hoveredEventId]];
-  const datasetRole = getDatasetRoleFromEventId(hoveredEventId);
-  const datasetName = getDatasetNameByRole(datasetRole);
+  const inferencesRole = getInferencesRoleFromEventId(hoveredEventId);
+  const inferencesName = getDatasetNameByRole(inferencesRole);
```
Comment (axiomofjoy) [inline app/src/components/pointcloud/PointCloudPointTooltip.tsx:46]: Missed `getDatasetNameByRole`

**Review by axiomofjoy — APPROVED** [review]

**Pull request merged**